### PR TITLE
Clean up sub_clean's cleanCleanfiles

### PR DESCRIPTION
**Pull request by cassella**
Correct and improve the comment (c is the file, not f),
and rename f to d since it's a directory.

### CODE REVIEW

**Comment by cassella** [comment]
Lightly tested by running start_test on a test with a `.cleanfiles`.

File: util/test/sub_clean [hunk 1]
```diff
@@ -53,19 +53,19 @@ def cleanChapelTest(f):
...
         cleanfile = c
     if os.access(cleanfile,os.R_OK):
         cleanfiles = ReadCleanfiles(cleanfile)
         if rmCore:
             cleanfiles += ['core']
         for file in cleanfiles:
-            if f:
-                globfiles = glob.glob(f+'/'+file)
+            if d:
+                globfiles = glob.glob(d+'/'+file)
```
Comment (ronawho) [inline util/test/sub_clean]: Any interest in switching things like `d+'/'+file` over to `os.path.join(d, file)` while you're here?

File: util/test/sub_clean [hunk 2]
```diff
@@ -53,19 +53,19 @@ def cleanChapelTest(f):
...
+def cleanCleanfiles(d, c, rmCore):
+    if d:
+        cleanfile = d+'/'+c
     else:
         cleanfile = c
     if os.access(cleanfile,os.R_OK):
         cleanfiles = ReadCleanfiles(cleanfile)
         if rmCore:
             cleanfiles += ['core']
         for file in cleanfiles:
```
Comment (ronawho) [inline util/test/sub_clean]: Maybe `file` => `f` since `file` gets highlighed and is kinda reserved?

**Comment by ronawho** [comment]
Nice, looks good. I left a few inline comments for other potential cleanups if you're interested in making those while you're here. If not, I'm happy to merge as-is

**Comment by cassella** [comment]
Ok, those changes are incoming.

**Comment by ronawho** [comment]
Ah, I see what you were talking about with some of the calls to glob not making sense with os.path.join. You could replace the remaining `/`'s with `os.path.sep`, but I don't feel strongly about that.

This looks great to me, and testing passed (I just did a full paratest and ran `start_test --clean-only` to make sure the same files were being cleaned up w/ and w/o this PR.)

I'll merge tomorrow.

**Comment by cassella** [comment]
I eventually decided the join call wouldn't be worse than the `d+"/"+file`, whatever `d` and `file` are.

Note, at least as far as the globbing is concerned, the globs won't match anything if there aren't files left over from running the tests once first.

**Comment by cassella** [comment]
Oh, I think I was reading too quickly.  The `--clean-only` was after the paratest, not how you ran it.  Never mind that last part.

**Pull request merged**